server/src/main/java/io/crate/planner/operators/Distinct.java

```diff
@@ -31,11 +31,15 @@
 
 public final class Distinct {
 
-    public static LogicalPlan create(LogicalPlan source, boolean distinct, List<Symbol> outputs, TableStats tableStats) {
+    public static LogicalPlan create(LogicalPlan source,
+                                     long numDocs,
```

Comment:
Having `numDocs` as parameter is not great, but i don't want to propagate the StatsProvider outside of the Optimizer/Rule Scope.